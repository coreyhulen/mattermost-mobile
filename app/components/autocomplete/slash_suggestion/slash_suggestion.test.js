// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';
import {
    FlatList,
} from 'react-native';

import SlashSuggestion from './slash_suggestion';

describe('SlashSuggestion', () => {
    const baseProps = {
        actions: {
            getAutocompleteCommands: jest.fn(),
        },
        fuse: {},
        theme: {},
        onChangeText: jest.fn(),
        onResultCountChange: jest.fn(),
        currentTeamId: '',
        isLandscape: false,
    };

    test('should render component without error', () => {
        const wrapper = shallow(
            <SlashSuggestion {...baseProps}/>,
        );
        expect(wrapper.type()).toEqual(null);
    });

    test('fire onResultCountChange when slash command was not found', () => {
        const wrapper = shallow(
            <SlashSuggestion {...baseProps}/>,
        );
        const props2 = {
            value: 'AðŸ‡ºðŸ‡¸Z',
        };
        wrapper.setProps(props2);
        expect(baseProps.onResultCountChange).toHaveBeenCalled();
        expect(baseProps.onResultCountChange).toHaveBeenLastCalledWith(0);
    });

    test('fire appropriate function when slash command was found', () => {
        const wrapper = shallow(
            <SlashSuggestion {...baseProps}/>,
        );
        wrapper.instance().handleFuzzySearch = jest.fn();
        const props2 = {
            value: '/awa',
            commands: [{auto_complete: true, display_name: 'away'}],
            currentTeamId: 'team1',
        };
        wrapper.setProps(props2);
        expect(baseProps.actions.getAutocompleteCommands).toHaveBeenCalledTimes(1);
        expect(baseProps.actions.getAutocompleteCommands).toHaveBeenLastCalledWith('team1');
        expect(wrapper.find(FlatList).exists()).toBe(true);
        expect(baseProps.onResultCountChange).toHaveBeenLastCalledWith(1);
    });
});
