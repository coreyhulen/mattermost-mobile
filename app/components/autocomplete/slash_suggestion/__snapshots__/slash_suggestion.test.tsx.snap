// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/autocomplete/slash_suggestion should match snapshot 1`] = `
<FlatList
  data={
    Array [
      Object {
        "Complete": "thetrigger",
        "Description": "The Description",
        "Hint": "The Hint",
        "IconData": "iconurl.com",
        "Suggestion": "/thetrigger",
      },
    ]
  }
  disableVirtualization={false}
  extraData={
    Object {
      "active": true,
      "dataSource": Array [
        Object {
          "Complete": "thetrigger",
          "Description": "The Description",
          "Hint": "The Hint",
          "IconData": "iconurl.com",
          "Suggestion": "/thetrigger",
        },
      ],
      "lastCommandRequest": 1234,
    }
  }
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  keyboardShouldPersistTaps="always"
  maxToRenderPerBatch={10}
  nestedScrollEnabled={false}
  numColumns={1}
  onEndReachedThreshold={2}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={50}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
        "flex": 1,
        "paddingTop": 8,
      },
      Object {
        "maxHeight": 50,
      },
    ]
  }
  testID="slash_suggestion.list"
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
